contracts/balancer/IndexPool.sol

```diff
@@ -919,19 +788,20 @@ contract IndexPool is BToken, BMath, IIndexPool {
     _pullUnderlying(tokenIn, msg.sender, tokenAmountIn);
     _pushUnderlying(tokenOut, msg.sender, tokenAmountOut);
 
-    // Update the balance and (if necessary) weight of the input token.
-    realInBalance = badd(realInBalance, tokenAmountIn);
-    _updateInputToken(tokenIn, inRecord, realInBalance);
-    if (inRecord.ready) {
-      inRecord.balance = realInBalance;
-    }
     // Update the in-memory record for the spotPriceAfter calculation,
     // then update the storage record with the local balance.
     outRecord.balance = bsub(outRecord.balance, tokenAmountOut);
     _records[tokenOut].balance = outRecord.balance;
     // If needed, update the output token's weight.
     _decreaseDenorm(outRecord, tokenOut);
 
+    // Update the balance and (if necessary) weight of the input token.
+    realInBalance = badd(realInBalance, tokenAmountIn);
+    _updateInputToken(tokenIn, inRecord, realInBalance);
+    if (inRecord.ready) {
+      inRecord.balance = realInBalance;
+    }
+
```

Comment:
Why the reorder here?

Reply:
The weight update will not execute if the total denormalized weight would exceed the maximum (27e18). The order was changed so that the weight decrease occurs first, leaving more room for the weight increase to execute.